contracts/interchain-token-service/tests/deploy_remote_canonical_token.rs

```diff
@@ -62,8 +62,35 @@ fn deploy_remote_canonical_token_succeeds() {
     }
     .abi_encode(&env);
 
+    let transfer_token_auth = mock_auth!(
+        env,
+        spender,
+        gas_token.transfer(spender, gas_service.address, gas_token.amount)
+    );
+
+    let pay_gas_auth = mock_auth!(
+        env,
+        spender,
+        gas_service.pay_gas(
+            client.address,
+            its_hub_chain,
+            its_hub_address,
+            payload,
+            spender,
+            gas_token,
+            Bytes::new(&env)
+        ),
+        &[(transfer_token_auth.invoke).clone()]
```

Comment:
```suggestion
        transfer_token_auth
```
let the macro take a variable number of sub invokes at the end and internally convert it to the above value. the usage could also allow accepting `[transfer_token_auth]` perhaps

Reply:
this could bring more complexity because of the upgrader tests, I'd rather revisit later if we need to take a variable with number of sub invokes and convert it to the value